src/providers/http.ts

```diff
@@ -43,14 +53,17 @@ export class HttpProvider implements ApiProvider {
     // Render all nested strings
     const nunjucks = getNunjucksEngine();
     const stringifiedConfig = safeJsonStringify(this.config);
-    const renderedConfig: { method: string; headers: Record<string, string>; body: any } =
-      JSON.parse(
-        nunjucks.renderString(stringifiedConfig, {
-          prompt,
-          ...context?.vars,
-        }),
-      );
-
+    const renderedConfigString = nunjucks.renderString(stringifiedConfig, {
+      // nunjucks does not escape JSON strings, so we need to escape them manually
+      prompt: isValidJson(prompt) ? prompt.replace(/"/g, '\\"') : prompt,
```

Comment:
`JSON.stringify(output).slice(1, -1)` should be a more foolproof way to do this (this is what we do for outputs in `matchers.ts`) because it will handle newlines, tabs, and backslashes too.

Reply:
thank you!